File: event-schemas/src/entities/Advertisement.ts


class Advertisement {
  private _publisher?: string;
  private _position?: number;
  private _totalLength?: number;
  private _quartile?: number;
  private _placementId?: string;
  private _placementName?: string;
  private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null>;

  constructor(input: {
    publisher?: string,
    position?: number,
    totalLength?: number,
    quartile?: number,
    placementId?: string,
    placementName?: string,
  }) {
    this._publisher = input.publisher;
    this._position = input.position;
    this._totalLength = input.totalLength;
    this._quartile = input.quartile;
    this._placementId = input.placementId;
    this._placementName = input.placementName;
  }

  get publisher(): string | undefined { return this._publisher; }
  set publisher(publisher: string | undefined) { this._publisher = publisher; }

  get position(): number | undefined { return this._position; }
  set position(position: number | undefined) { this._position = position; }

  get totalLength(): number | undefined { return this._totalLength; }
  set totalLength(totalLength: number | undefined) { this._totalLength = totalLength; }

  get quartile(): number | undefined { return this._quartile; }
  set quartile(quartile: number | undefined) { this._quartile = quartile; }

  get placementId(): string | undefined { return this._placementId; }
  set placementId(placementId: string | undefined) { this._placementId = placementId; }

  get placementName(): string | undefined { return this._placementName; }
  set placementName(placementName: string | undefined) { this._placementName = placementName; }

  get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null> | undefined { return this._additionalProperties; }
  set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null> | undefined) { this._additionalProperties = additionalProperties; }
}
export default Advertisement;

File: event-schemas/data/csharp/Advertisement.cs
namespace entities
{
  using System.Collections.Generic;

  public class Advertisement
  {
    private string publisher;
    private double? position;
    private double? totalLength;
    private double? quartile;
    private string placementId;
    private string placementName;
    private Dictionary<string, dynamic> additionalProperties;

    public string Publisher 
    {
      get { return publisher; }
      set { publisher = value; }
    }

    public double? Position 
    {
      get { return position; }
      set { position = value; }
    }

    public double? TotalLength 
    {
      get { return totalLength; }
      set { totalLength = value; }
    }

    public double? Quartile 
    {
      get { return quartile; }
      set { quartile = value; }
    }

    public string PlacementId 
    {
      get { return placementId; }
      set { placementId = value; }
    }

    public string PlacementName 
    {
      get { return placementName; }
      set { placementName = value; }
    }

    public Dictionary<string, dynamic> AdditionalProperties 
    {
      get { return additionalProperties; }
      set { additionalProperties = value; }
    }
  }
}